### PR TITLE
Render PDFs using pdf_viewer.js

### PR DESCRIPTION
**Pull request by d2l-joelp**
[US64311]: [Make alternate PDF rendering (pdf.js) accessible and readable by screen readers](https://rally1.rallydev.com/#/44847733393d/detail/userstory/48726566270)

Switch to using pdf_viewer.js instead of the lower level pdf.js (viewer is built on top of pdf.js) to get the following benefits:
- Accessible content, consumable by screen readers (actual content in the DOM instead of just stuff rendered in a canvas)
- Selectable text
- Searchable text (via ctrl-F in the browser)
- Spinners while loading pages
- More advanced and memory efficient page loading/unloading
- Much less code
- Easier base from which to build a viewer UI (i.e. next/prev, go to page, zoom, etc.)

Load pdf.worker.js from cross domain CDN

### CODE REVIEW

File: src/plugins/pdf/alternate/alternateViewer.js [hunk 1]
```diff
@@ -2,123 +2,61 @@
...
 		src: React.PropTypes.string.isRequired,
-		progressCallback: React.PropTypes.func,
-		maxScale: React.PropTypes.number
+		progressCallback: React.PropTypes.func
 	},
-	getMaxScale: function() {
-		return this.props.maxScale || MAX_SCALE;
+	container: null,
+	pdfLinkService: null,
+	pdfViewer: null,
```
Comment (jstefaniuk-d2l) [inline src/plugins/pdf/alternate/alternateViewer.js]: I think for the most part we've been trying to keep variables like these in the component's `state`

Comment (d2l-joelp) [inline src/plugins/pdf/alternate/alternateViewer.js]: I shied away from setting state only because I didn't want to trigger a render or have to implement shouldComponentUpdate to prevent it. I like your idea of passing the values into loadDocument and if I can still access to the DOM node in componentWillUnmount then I don't need the container variable anymore either.

File: README.md [hunk 2]
```diff
@@ -11,11 +11,18 @@ Install NPM dependencies:
...
 ```
 
-Additionally, to get the fallback (non-native) PDF renderer working, you will need to add a reference to pdf.js from the CDN.
-It must be referenced _before_ referencing the javascript file containing the code for this component.
+Additionally, to get the fallback (non-native) PDF renderer working, you will need to reference some dependencies from the CDN.
+These must be referenced _before_ referencing the css and js files containing the code for this component.
+
+```css
+<link rel="stylesheet" href="https://s.brightspace.com/lib/pdf.js/1.2.109/pdf_viewer.css">
+<link rel="stylesheet" type="text/css" href="dist/app.css">
```
Comment (CodeBaboon) [inline README.md]: If the user just bower and npm installs, do they have a `dist/` dir?

Comment (d2l-joelp) [inline README.md]: No, there would be no dist dir and currently we are only consuming this as a react component and assuming Browserify to resolve modules and frau-sass-importer to resolve scss. Perhaps we could /should improve the instructions overall and maybe do it in a separate PR.

File: src/plugins/pdf/alternate/__tests__/isCrossDomainTests.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+	it('should be true when host names differ', function() {
+		expect(isCrossDomain('https://www.brightspace.com/lib/pdf.js/pdf.worker.js')).toEqual(true);
+	});
+
+	it('should be true when ports differ', function() {
+		expect(isCrossDomain('https://s.brightspace.com:1234/lib/pdf.js/pdf.worker.js')).toEqual(true);
+	});
+
+	it('should be false when url is not a string', function() {
```
Comment (CodeBaboon) [inline src/plugins/pdf/alternate/__tests__/isCrossDomainTests.js]: Could do the foreach trick here:

```
[1, [], {}].forEach(function(input) {
  it('should be false when url is not a string', function() {
    expect(isCrossDomain(1)).toEqual(false);
    expect(isCrossDomain([])).toEqual(false);
    expect(isCrossDomain({})).toEqual(false);
  });
};
```

**Comment by jstefaniuk-d2l** [comment]
I see no other issues so LGTM aside from whatever Dave points out

File: src/plugins/pdf/alternate/alternateViewer.js [hunk 4]
```diff
@@ -2,123 +2,64 @@
...
 var React = require('react'),
 	pdfjs = require('./pdfjs-lib'),
-	AlternateViewerPage = require('./alternateViewerPage'),
-	isInView = require('./isInView'),
-	getPixelRatio = require('./pixelRatio/getPixelRatio');
-
-var MAX_SCALE = 1.5;
+	pdfjsWorkerSrcInit = require('./pdfjsWorkerSrcInit');
 
 var AlternativeViewer = React.createClass({
```
Comment (jstefaniuk-d2l) [inline src/plugins/pdf/alternate/alternateViewer.js]: This class should have a getInitialState method containing all of the state variables

File: src/plugins/pdf/alternate/alternateViewer.js [hunk 5]
```diff
@@ -2,123 +2,64 @@
...
 			<div className="vui-fileviewer-pdf-alternate">
-				{self.state.pages.map(function(page) {
-					return <AlternateViewerPage key={page.ref} ref={page.ref}
-						page={page}
-						pixelRatio={self.state.pixelRatio}
-						scale={self.state.scale}
-						pageHeight={self.state.pageHeight}
-						pageWidth={self.state.pageWidth} />;
-				})}
+				<div ref="viewer" className="pdfViewer"></div>
```
Comment (CodeBaboon) [inline src/plugins/pdf/alternate/alternateViewer.js]: Do we use this ref anywhere?

Comment (d2l-joelp) [inline src/plugins/pdf/alternate/alternateViewer.js]: Nope, good catch. I was using it at one point, should've remove this.

**Comment by CodeBaboon** [comment]
LGTM

**Comment by Surekh** [comment]
LGTM! Known issues with Safari and Chrome will be addressed later.
